core-blocks/button/index.js

```diff
@@ -77,11 +86,20 @@ class ButtonBlock extends Component {
 			url,
 			title,
 			align,
-			color,
-			textColor,
 			clear,
 		} = attributes;
 
+		const textColor = initializeColor( {
```

Comment:
This `intiailizeColor` calls are bothering me a bit. It seems too magic to me. I wonder if it's not better to transform the `withColors` HoC to something like 

```
withColors( ( getColor, { attributes } ) => ( {
   backgroundColor: getColor( attributes.backgroundColor, attributes.customBackgroundColor, 'background-color' )
} )
```

Reply:
Thank you for the review and for the feedback @youknowriad. I like your suggestion, I will give it a try in a separate PR.
Regarding the setters (the logic to set the name attribute or the custom one) do you think we should pass a getColor to retrieve the data and a setColor to build the setter, or getColor returns the setter inside similar to what initiailizeColor does?

Reply:
I'd lean towards a separate `setColor`